File: master/Main.master.vb

Partial Class master_Main
    Inherits System.Web.UI.MasterPage

    Private _pointer As Boolean = False

    Protected Sub Page_Init(ByVal sender As Object, ByVal e As System.EventArgs) Handles Me.Init
        'Session("gstrUserID") = "800017" 'user=62502, approver=800017
        'Session("gettemp") = "admin"
        'Session("gstrUsername") = "admin"
        'Session("gstrUserCompCode") = "System"
        'Session("gstrUserWorksNo") = "System"
    End Sub

    Protected Sub Page_Load(ByVal sender As Object, ByVal e As System.EventArgs) Handles Me.Load

        If Request.QueryString(ACL.Control.URL.URLEMPLOYEEID) IsNot Nothing Then
            Session("gstrUserID") = ACL.Security.Encryption.Decrypt(Request.QueryString(ACL.Control.URL.URLEMPLOYEEID))
            _pointer = True
        End If

        If Request.QueryString(ACL.Control.URL.URLCOMPANYID) IsNot Nothing Then
            Session("gstrUserCompCode") = Request.QueryString(ACL.Control.URL.URLCOMPANYID)
            _pointer = True
        End If

        If Session("gstrUserID") Is Nothing Then
            If Request.UserAgent.ToLower().Contains("ipad") Then
                Response.Redirect("~/SessionExpired.aspx?ID=1")
            End If
            Response.Redirect("~/SessionExpired.aspx?ID=2")
        End If
    End Sub

End Class

File: master/Main.master.cs
using System;

public partial class master_Main : System.Web.UI.MasterPage
{
    private bool _pointer = false;

    protected void Page_Init(object sender, EventArgs e)
    {
    }

    protected void Page_Load(object sender, EventArgs e)
    {
        if (Request.QueryString[ACL.Control.URL.URLEMPLOYEEID] != null)
        {
            Session["gstrUserID"] = ACL.Security.Encryption.Decrypt(Request.QueryString[ACL.Control.URL.URLEMPLOYEEID]);
            _pointer = true;
        }

        if (Request.QueryString[ACL.Control.URL.URLCOMPANYID] != null)
        {
            Session["gstrUserCompCode"] = Request.QueryString[ACL.Control.URL.URLCOMPANYID];
            _pointer = true;
        }

        if (Session["gstrUserID"] == null)
        {
            if (Request.UserAgent.ToLower().Contains("ipad"))
            {
                Response.Redirect("~/SessionExpired.aspx?ID=1");
            }
            Response.Redirect("~/SessionExpired.aspx?ID=2");
        }
    }
}
